### PR TITLE
Replace log4cpp with boost.log

### LINKED ISSUE #45: use boost.log instead of log4cpp
I just discovered that boost got a [logging module in version 1.54](http://www.boost.org/doc/libs/1_54_0/libs/log/doc/html/index.html)

This would allow us to get rid of the log4cpp dependency on the cost of requiring at least debian jessie, as wheezy ships with boost 1.49.

But as the [libboost-log.54-dev package](https://packages.debian.org/jessie/libboost1.54-dev) is still "not available" in jessie, I think we should still wait a few months before implementing / releasing this change.

### PR DESCRIPTION
**Pull request by florianjacob**
Resolves #45.

### CODE REVIEW

**Comment by florianjacob** [comment]
@Semoar could you please give all of this a test on your server? Currently alacarte can't be compiled on Arch Linux due to other problems, somehow we're incompatible to either the current gcc or boost, so the only tests I could do is to see whether it compiles on travis ci.

**Comment by florianjacob** [comment]
Of course, all code reviews besides tests are appreciated, too. :wink: Touched many areas of the code I never had anything to do with before, there may be any specifics I've overlooked.

**Comment by alex85k** [comment]
Can build&run on Ubuntu, FreeBSD, CentOS VMs but have no idea how to check correctness :)

**Comment by Semoar** [comment]
Sorry, I have only Debian Wheezy on my servers, whose boost is too old.

**Comment by florianjacob** [comment]
@alex85k build&run is exactly what I had in mind, just to check that nothing went completely wrong. ;)
Pushed the new changes based on your new merged commits now, let's see what travis says. The conflict resolution wasn't as trivial as I hoped.

**Comment by alex85k** [comment]
Thank you for merging, the Git tree now looks extremely impressive :)

Will run the tests and trivial server config on virtual machines tomorrow.

**Comment by alex85k** [comment]
- Does log4cpp initialization make any sense in tests now? 
https://github.com/alacarte-maps/alacarte/blob/master/tests/server/server_test_module.cpp
https://github.com/alacarte-maps/alacarte/blob/master/tests/importer/importer_test_module.cpp
(travis stops without log4cpp now)

 - When building this branch on FreeBSD with manually installed Boost 1.59, I had the following error:
https://gist.github.com/alex85k/6aac3aa9367cb0fd79e1#file-alacarte-boost-log-log
Did they change something serions again in newest Boost?

**Comment by alex85k** [comment]
The same error on Windows Boost 1.59:
https://gist.github.com/alex85k/40d7e702394c13b033a7#file-alacarte-boostlog-error

File: src/alacarte_importer.cpp [hunk 1]
```diff
@@ -98,9 +94,9 @@ class AlacarteImporterApp
 
 		} catch(excp::InputFormatException& e)
 		{
-			log.errorStream() << "Failed to parse \"" << excp::ErrorOut<excp::InfoFileName>(e) << "\"!";
-			log.errorStream() << "Error:";
-			log.errorStream() << excp::ErrorOut<excp::InfoWhat>(e);
+			LOG_SEV(importer_log, error) << "Failed to parse \"" << excp::ErrorOut<excp::InfoFileName>(e) << "\"!";
+			LOG_SEV(importer_log, error) << "Error:";
+			LOG_SEV(importer_log, error) << excp::ErrorOut<excp::InfoWhat>(e);
```
Comment (alex85k) [inline src/alacarte_importer.cpp]: Here is the problematic place for new Boost.

Comment (alex85k) [inline src/alacarte_importer.cpp]: Some more operator<< in exceptions.cpp are needed? Or conversion to string...

**Comment by florianjacob** [comment]
Regarding the test cases, I completely forgot that there might be some logging stuff, too. Just pushed a commit to also make the changes in the tests.

Will look into the other issues soon. Seems like we have a number of problems with new boost, maybe there's a changelog and/or porting guide somewhere…? I'd like to address issues with new boost in another PR though, and get this merged as soon as it seems like there's no issue with 1.55 left.

**Comment by alex85k** [comment]
Agree . There is really a breaking change in boost 1.59 considering exceptions and type information: http://www.boost.org/users/history/version_1_59_0.html

I was able to build on FreeBSD 10.2, default boost 1.55, after fixing boost bug https://svn.boost.org/trac/boost/ticket/9468  manually and disabling -Werror (many deprecated `register` and some other warnings).

File: CMakeLists.txt [hunk 2]
```diff
@@ -163,17 +158,18 @@ IF(GMOCK_FOUND)
...
-target_link_libraries(unitTests_importer    ${Boost_LIBRARIES} ${Log4Cpp_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries(unitTests_server      ${Boost_LIBRARIES} ${Log4Cpp_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
-target_link_libraries(unitTests_parser      ${Boost_LIBRARIES} ${Log4Cpp_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(alacarte-server       ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${SYSTEM_LIBRARIES})
+target_link_libraries(alacarte-importer     ${Boost_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} ${SYSTEM_LIBRARIES})
+target_link_libraries(unitTests_utils       ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(unitTests_general     ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(unitTests_importer    ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(unitTests_server      ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```
Comment (alex85k) [inline CMakeLists.txt]: Could you add ${SYSTEM_LIBRARIES} here to make branch compileble on BSD?

File: CMakeLists.txt [hunk 3]
```diff
@@ -163,17 +158,18 @@ IF(GMOCK_FOUND)
...
+target_link_libraries(unitTests_parser      ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```
Comment (alex85k) [inline CMakeLists.txt]: This line wants ${SYSTEM_LIBRARIES} too...

File: CMakeLists.txt [hunk 4]
```diff
@@ -163,17 +158,18 @@ IF(GMOCK_FOUND)
...
 IF(GMOCK_FOUND)
-    target_link_libraries(unitTests_mapcss  ${Boost_LIBRARIES} ${Log4Cpp_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} gmock gtest)
+    target_link_libraries(unitTests_mapcss  ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT} gmock gtest)
 ENDIF(GMOCK_FOUND)
-target_link_libraries(unitTests_eval        ${Boost_LIBRARIES} ${Log4Cpp_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(unitTests_eval        ${Boost_LIBRARIES} ${CAIRO_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```
Comment (alex85k) [inline CMakeLists.txt]: And the last ${SYSTEM_LIBRARIES}, please :)

**Comment by alex85k** [comment]
@florianjacob 
Here is the testing output for FreeBSD, you can check the log messages formats if needed:
https://gist.github.com/alex85k/36838da1d2355bd11bd0
Seems to be working fine.

**Comment by alex85k** [comment]
Considering ``SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")`` - maybe you could put it into
```cmake
if(CMAKE_COMPILER_IS_GNUCXX)
   SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
endif()
```
? Standard boost.log headers give many warning on CLang...

**Comment by florianjacob** [comment]
@alex85k thanks for testing all of this and pasting the output for me to check. I added the fixes for clang and FreeBSD you mentioned. As Travis doesn't show up with anything unexpected, I'll merge this now, so we can continue from this stage with the other pending PRs.

**Pull request merged**